packages/chatbot-server-mongodb-public/src/tools/fetchPage.ts

```diff
@@ -0,0 +1,111 @@
+import { z } from "zod";
+import { FindContentFunc, MongoDbPageStore, Reference } from "mongodb-rag-core";
+import { normalizeUrl } from "mongodb-rag-core/dataSources";
+import { wrapTraced } from "mongodb-rag-core/braintrust";
+import { Tool, tool, ToolExecutionOptions } from "mongodb-rag-core/aiSdk";
+import { addReferenceSourceType } from "../processors/makeMongoDbReferences";
+
+export const FETCH_PAGE_TOOL_NAME = "fetch_page";
+
+export const MongoDbFetchPageToolArgsSchema = z.object({
+  pageUrl: z.string().describe("URL of page to fetch"),
+  query: z.string().describe("Search query"),
+});
+
+export type MongoDbFetchPageToolArgs = z.infer<
+  typeof MongoDbFetchPageToolArgsSchema
+>;
+
+export type FetchPageToolResult = {
+  url: string;
+  text: string;
+  reference?: Reference;
```

Comment:
per below comment about the search fall back, consider doing `references: References` (plural). then u can also remove the `url` field.